core/src/main/java/org/apache/druid/math/expr/vector/VectorProcessors.java

```diff
@@ -53,11 +53,17 @@ public static <T> ExprVectorProcessor<T> makeSymmetricalProcessor(
   {
     final ExpressionType leftType = left.getOutputType(inspector);
 
+    // if type is null, it means the input is all nulls
     if (leftType == null) {
       return right.buildVectorized(inspector);
     }
 
-    Preconditions.checkArgument(inspector.areSameTypes(left, right));
+    Preconditions.checkArgument(
+        inspector.areSameTypes(left, right),
+        "%s and %s are not the same type",
```

Comment:
Can we print the type name of left and right in the error message?

Reply:
this is printing the type names